core-rust/state-manager/src/state_manager.rs

```diff
@@ -1005,26 +1001,27 @@ where
 
             let commit_result = match &processed.receipt().result {
                 TransactionResult::Commit(result) => {
-                    if let Some((_, next_epoch)) = result.next_epoch {
+                    if result.next_epoch.is_some() {
                         let is_last = i == (commit_transactions_len - 1);
                         if !is_last {
                             return Err(CommitError::MissingEpochProof);
                         }
-                        // TODO: Use actual result and verify proof validator set matches transaction receipt validator set
-                        epoch_boundary = Some(next_epoch);
```

Comment:
_(note: Right now, we no longer have to "find" the `epoch_boundary`, since it does not go into `CommitBundle {...}` anymore, because the entire `proof` goes there [and it contains the `next_epoch` itself].)_